doc/index.rst

```diff
@@ -251,6 +251,29 @@ So either `ABICHECK_URL` or `ABICHECK_FALLBACK` must be set.
 The url should point to an archive (\*.tar.\*,\*.zip, \*.tgz or \*.tbz2). The file's basename will be displayed as version.
 As an alternative `ABICHECK_VERSION` can be provided explicitly.
 
+Example ABI checking
+++++++++++++++++++++
+
+Let's look at a few example configs in `.travis.yml in this repository, industrial_ci <https://github.com/ros-industrial/industrial_ci/blob/master/.travis.yml>`_. Note that all of these examples check the external repositories (i.e. not industrial_ci) specified by `_EXTERNAL_REPO`.
+
+Case-1: Check ABI compatibility between current and the specific version.
+
+```
+- ROS_DISTRO=indigo _EXTERNAL_REPO='ros/actionlib#38ce66e2ae2ec9c19cf12ab22d57a8134a9285be' ROS_REPO=ros ABICHECK=true
+```
+
+(TODO: Which branch and commit is going to be used for the check in this case?)
+
+Case-2: Check ABI compatibility between the specific 2 versions.
+
+```
+    - ROS_DISTRO=kinetic _EXTERNAL_REPO='ros-industrial/ros_canopen#0.7.6' ABICHECK=true ABICHECK_URL=https://github.com/ros-industrial/ros_canopen/archive/0.7.5.tar.gz
+```
+
+The version of the repo passed in `_EXTERNAL_REPO` ([ros-industrial/ros_canopen#0.7.5](https://github.com/ros-industrial/ros_canopen/releases/tag/0.7.5)) wiii be checked against the tarball of the version specified in `ABICHECK_URL`.
+
+
+
```

Comment:
Case 2: Check ABI compatibility for merges (e.g. Pull request):
```
- ROS_DISTRO=kinetic ABICHECK=true ABICHECK_FALLBACK=https://github.com/ros-industrial/ros_canopen/archive/kinetic-devel.tar.gz
```
In a pull request or for merge commits the ABI gets checked for the merged version against the first parent (e.g. target branch of the pull request).
However, for push/branch test jobs the HEAD commit might not be a merge;in this case the `ABICHECK_FALLBACK` will be used instead.